### ISSUE TITLE
[Plan] Add input/output testing

### ISSUE THREAD

**Issue by ChrisJefferson** [gap-system/gap#371]
TL;DR: We can't test the break loop. We'll test it by feeding a text file into GAP and diffing it against expected output.

Longer: 

We can't currently test anything involving the debug loop. The two main problems this causes is:

1) `Where`,`UpEnv`,`ShowMethods` other break-loop specific functions can break
2) Backtraces can be inaccurate.

I plan to fix this in the (hopefully) simplest way possible, by adding a directory of special tests of the form:

`testfile` : a keyboard input for GAP
`testfile.stdout` : The output which should appear on stdout.
`testfile.errout` : Output which should appear on stderr.

These will be run as follows (we have to be slightly careful, because GAP tries to grab the terminal for reading input for break loops. Doing `>file 2>otherfile` both catches things written to stderr and stops this behaviour).

```
gap.sh < testfile > testfile.stdoutT 2>testfile.erroutT -b; diff testfile.stdout testfile.stdoutT && diff testfile.errout testfile.erroutT
```

These files are going to look fairly horrible, for example given a nice short input:

```
test:= function()
  Error("cheese");
end;;

test( );
quit;
```

We get stdout: 

```
gap> > > gap> gap> <function "test">( <arguments> )
 called from read-eval loop at line 5 of *stdin*
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
```

And errout: `Error, cheese called from`

I considered using something like expect, but the advantage of this "format" is that it will be easy to regenerate all output files when we tweak how output occurs.

**Comment by fingolfin** [comment]
Sounds like a good plan.

**Comment by alex-konovalov** [comment]
@ChrisJefferson can this now be closes as we have tst/test-error (for which I have a reminder in #1090 to make it running regularly)?
